docs/DesignDocs/InitTimeConfigurationInterface.md

```diff
@@ -0,0 +1,717 @@
+ * endorsements buffer in bytes.
+ * @retval OE_OK The function succeeded.
+ * @retval OE_INVALID_PARAMETER At least one of the parameters is invalid.
+ * @retval OE_NOT_FOUND The input evidence format id is not supported.
+ * @retval other appropriate error code.
+ */
+oe_result_t oe_get_evidence(
+    const oe_uuid_t* format_id,
+    uint32_t flags,
+    const void* custom_claims_buffer,
+    size_t custom_claims_buffer_size,
+    const void* optional_parameters,
+    size_t optional_parameters_size,
+    uint8_t** evidence_buffer,
+    size_t* evidence_buffer_size,
+    uint8_t** endorsements_buffer,
+    size_t* endorsements_buffer_size);
+```
+
+The following pseudocode demonstrates how the proposed interface can be used. 
+```C
+
+    /// Untrusted Code of the OE Application
+    ///
+
+    // Set up extra content and configid
+    rsa3072_public_key_t  pubkey = {...};
+    oe_sgx_config_data_t = 
+    {
+        SHA256(pubkey), // configid
+        0
+    };
+    oe_enclave_setting_t settings[] =
+    {
+        {
+            setting_type = OE_SGX_CONFIG_DATA,
+            u.oe_sgx_config_data_t = &oe_sgx_config_data_t
+        }
+    };
+    // Create enclave
+    oe_create_enclave(..., settings, ...);
+
+    /// Enclave code of the OE Application
+    ///
+
+    // Load the pubkey into enclave memory and verify it against
+    // configid
+    ...
+
+    // Set up init-time claims for attestation
+    uint8 claim_buffer[sizeof(uint32_t)+sizeof(rsa3072_public_key_t);
+    oe_inittime_custom_claim_buffer_t *p_inittime_custom_claims_buffer
+                                                        = claim_buffer;
+
+    p_inittime_custom_claims_buffer->integrity_algorithm_id = 0;
+    memcpy(p_inittime_custom_claims_buffer->buffer, pubkey,
+           sizeof(rsa3072_public_key_t));
+  
+    size_t inittime_custom_claims_buffer_size =
+        sizeof(claim_buffer);
+
+    // Generate evidence #1
+    oe_get_evidence(...,
+        runtime_custom_claims_buffer_1,
+        runtime_custom_claims_buffer_size_1,
+        evidence_buffer_1,
+        ...);
+
+    //Assemble the data buffer with evidence and inittime_claim
+    ...
+    data_buffer->evidence_buffer_size = evidence_buffer_size_1;
+    data_buffer->inittime_claims_buffer_size = inittime_custom_claims_buffer_size;
+    memcpy(data_buffer->data, evidence_buffer_1, evidence_buffer_size_1);
+    memcpy(data_buffer->data + evidence_buffer_size_1;
+       p_inittime_custom_claims_buffer, inittime_custom_claims_buffer_size);
+    ...
+
+    // Send data_buffer to Verifier
+    send_evidence_to_REST_URL(..., data_buffer, data_buffer_size, ....);
+    ...
+
+    // Generate evidence #2
+    oe_get_evidence(...,
+        runtime_custom_claims_buffer_2,
+        runtime_custom_claims_buffer_size_2,
+        p_inittime_custom_claims_buffer,
+        inittime_custom_claims_buffer_size,
+        evidence_buffer_2,
+        ...);
```

Comment:
That's for alternative 1 right? where `inittime_custom_claims_buffer` is not passed to `oe_get_evidence`?